lib/src/extractLevels.js

```diff
@@ -0,0 +1,191 @@
+ * @param {array} elementArray Extracted related element arrays from IFC file
+ * @return {array} groupedElements
+ */
+function groupElements(elementArray) {
+  const groupedElements = []
+  for (let i = 0; i<elementArray.length+1; i++) {
+    let joinedArray = []
+    joinedArray = elementArray
+    joinedArray.splice(i, 1)
+    const flatElements = joinedArray.flat()
+    groupedElements[i] = flatElements
+  }
+  return groupedElements
+}
+
+
+/**
+ * Create a set of new IFC files and remove related elements fot each level
+ * @param {string} inputIfcFilename
+ * @param {array} relatedElemGrouped Express IDs of grouped related
+ *     elements for each level
+ */
+function createManipulateNewIFC(inputIfcFilename, relatedElemGrouped) {
+  for (let i = 0; i < relatedElemGrouped.length; i++) {
+    const newfilename = `${inputIfcFilename}_Level-${i}.ifc`
+    copyFiles(inputIfcFilename, newfilename, relatedElemGrouped[i])
+  }
+}
+
+
+/**
+ * Create map to find Line numbers based on expressID
+ * @param {string} IFCdata - Input raw IFC data
+ * @return {Map} Map with Express IDs as keys
+ */
+function expressIDtoLineMap(IFCdata) {
+  const expressIDMaps = new Map()
+  const lines = IFCdata.split('\n')
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i]
+    {
+      if ((line.indexOf('#') != -1) && (line.indexOf('=') != -1)) {
+        const data1 = line.split('#')
+        const data2 = data1[1].split('=')
+        const expressID = data2[0]
+        expressIDMaps.set(parseInt(expressID), i)
+      }
+    }
+  }
+  return expressIDMaps
+}
+
+
+/**
+ * Find lines based on an array of express IDs
+ * @param {string} IFCdata Input raw IFC data.
+ * @param {array} expressIDs Express IDs to be removed.
+ * @return {array} list of lines corresponding to the input array of
+ *     expressIDs.
+ */
+function findLinesWithExpressID(IFCdata, expressIDs = []) {
+  const expressIDMap = expressIDtoLineMap(IFCdata)
+  const lineIndexes = []
+  for (let i = 0; i < expressIDs.length; i++) {
+    try {
+      lineIndexes[i] = expressIDMap.get(expressIDs[i])
+    } catch (e) {
+      // Hmm? void(0)
+      console.error(e)
+    }
+  }
+  return lineIndexes
+}
+
+
+const removeLines = (data, lines = []) => {
+  return data
+      .split('\n')
+      .filter((val, idx) => lines.indexOf(idx) === -1)
+      .join('\n')
+}
+
+
+/**
+ * Remove lines from an IFC file with an array of expressIDs
+ * @param {string} filename - filename of new IFC file to remove elements
+ * @param {array} expressIDsToRemove - array of expressIDs to remove
+ */
+function removeLinesFromIFCwithExpressID(filename, expressIDsToRemove) {
+  const fileCopiedBool = true
+  if (fileCopiedBool) {
+    fs.readFile(filename, 'utf8', (err, data) => {
+      if (err) throw err
+      const linesToRemove = findLinesWithExpressID(data, expressIDsToRemove)
+      // console.log(`Lines to remove from ${filename} are ${linesToRemove}`)
```

Comment:
remove debug comments, here and throughout

Reply:
Removed console.log that were commented out